diff --git a/options/options.c b/options/options.c
--- a/options/options.c
+++ b/options/options.c
@@ -317,6 +317,9 @@ const struct m_sub_options mp_subtitle_sub_opts = {
         {"sub-gray", OPT_BOOL(sub_gray)},
         {"sub-ass", OPT_BOOL(ass_enabled), .flags = UPDATE_SUB_HARD},
         {"sub-scale", OPT_FLOAT(sub_scale), M_RANGE(0, 100)},
+        {"image-subs-scale", OPT_FLOAT(image_subs_scale), M_RANGE(0, 100)},
+        {"image-subs-scale-x", OPT_FLOAT(image_subs_scale_x), M_RANGE(0, 100)},
+        {"image-subs-scale-y", OPT_FLOAT(image_subs_scale_y), M_RANGE(0, 100)},
         {"sub-scale-signs", OPT_BOOL(sub_scale_signs)},
         {"sub-line-spacing", OPT_FLOAT(sub_line_spacing), M_RANGE(-1000, 1000)},
         {"sub-ass-line-spacing", OPT_REPLACED("sub-line-spacing")},
@@ -367,6 +370,9 @@ const struct m_sub_options mp_subtitle_sub_opts = {
         .ass_prune_delay = -1.0,
         .teletext_page = 0,
         .sub_scale = 1,
+        .image_subs_scale = 1,
+        .image_subs_scale_x = 1,
+        .image_subs_scale_y = 1,
         .ass_vsfilter_color_compat = 1,
         .ass_use_video_data = 2,
         .ass_video_aspect = 0,
diff --git a/options/options.h b/options/options.h
--- a/options/options.h
+++ b/options/options.h
@@ -95,6 +95,9 @@ struct mp_subtitle_opts {
     bool ass_scale_with_window;
     struct osd_style_opts *sub_style;
     float sub_scale;
+    float image_subs_scale;
+    float image_subs_scale_x;
+    float image_subs_scale_y;
     bool sub_scale_signs;
     float sub_gauss;
     bool sub_gray;
diff --git a/sub/sd_lavc.c b/sub/sd_lavc.c
--- a/sub/sd_lavc.c
+++ b/sub/sd_lavc.c
@@ -485,18 +485,27 @@ static struct sub_bitmaps *get_bitmaps(struct sd *sd, struct mp_osd_res d,
 
     osd_rescale_bitmaps(res, w, h, d, video_par);
 
-    if (opts->sub_scale != 1.0 && shared_opts->ass_style_override[sd->order]) {
-        for (int n = 0; n < res->num_parts; n++) {
-            struct sub_bitmap *sub = &res->parts[n];
-
-            float shit = (opts->sub_scale - 1.0f) / 2;
-
-            // Fortunately VO isn't supposed to give a FUCKING FUCK about
-            // whether the sub might e.g. go outside of the screen.
-            sub->x -= sub->dw * shit;
-            sub->y -= sub->dh * shit;
-            sub->dw += sub->dw * shit * 2;
-            sub->dh += sub->dh * shit * 2;
+    if (shared_opts->ass_style_override[sd->order]) {
+        if (opts->image_subs_scale != 1.0 || opts->image_subs_scale_x != 1.0 ||
+            opts->image_subs_scale_y != 1.0) {
+            for (int n = 0; n < res->num_parts; n++) {
+                struct sub_bitmap *sub = &res->parts[n];
+
+                float shit_x, shit_y;
+                if (opts->image_subs_scale_x == 1.0 && opts->image_subs_scale_y == 1.0) {
+                    shit_x = shit_y = (opts->image_subs_scale - 1.0f) / 2;
+                } else {
+                    shit_x = (opts->image_subs_scale_x - 1.0f) / 2;
+                    shit_y = (opts->image_subs_scale_y - 1.0f) / 2;
+                }
+
+                // Fortunately VO isn't supposed to give a FUCKING FUCK about
+                // whether the sub might e.g. go outside of the screen.
+                sub->x -= sub->dw * shit_x;
+                sub->y -= sub->dh * shit_y;
+                sub->dw += sub->dw * shit_x * 2;
+                sub->dh += sub->dh * shit_y * 2;
+            }
         }
     }
 